.github/workflows/ci.yml

```diff
@@ -0,0 +1,27 @@
+name: C/C++ CI
+
+on:
+  pull_request:
+    branches: [ develop ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v2
+    - name: Initialize submodules
+      run: git submodule update --init
+    - name: Make build directory
+      run: mkdir build
+    - name: Run cmake
+      run: cmake ../
+      working-directory: build
+    - name: Run make
+      run: make -j
+      working-directory: build
+    - name: Run tests
+      run: make test
+      working-directory: build
```

Comment:
I would add another step for deletion of artifacts and files that this process creates. You could probably do a simple rm -rf for now, or if you want to be fancy there are options for dealing with artifacts like artifacts:expire_in : time in seconds before an artifact’s deletion ( https://docs.gitlab.com/ee/ci/pipelines/job_artifacts.html ).

Reply:
So github's process may be different than gitlab. But in either case I think everything in the working directory is deleted by default (including the cloned repository and it's subdirectories, which would include the created build directory). I think you specify artifacts if you want things to stay around. But just to be safe I added a remove of the build directory.